Video-Transformers/Video-Transformers/Video-Transformers/OpenTokWrapper.m

```diff
@@ -0,0 +1,429 @@
+    return;
+  }
+  
+  #ifdef CONSOLE_LOGGING
+    otc_log_set_logger_callback(on_otc_log_message);
+    otc_log_enable(OTC_LOG_LEVEL_ALL);
+  #endif
+  
+  struct otc_session_callbacks session_callbacks = {0};
+  session_callbacks.user_data = session_data;
+  session_callbacks.on_connected = on_session_connected;
+  session_callbacks.on_connection_created = on_session_connection_created;
+  session_callbacks.on_connection_dropped = on_session_connection_dropped;
+  session_callbacks.on_stream_received = on_session_stream_received;
+  session_callbacks.on_stream_dropped = on_session_stream_dropped;
+  session_callbacks.on_disconnected = on_session_disconnected;
+  session_callbacks.on_error = on_session_error;
+  
+  session_data->session = otc_session_new(API_KEY, SESSION_ID, &session_callbacks);
+  
+  if (session_data->session == NULL) {
+    NSLog(@"Could not create OpenTok session successfully");
+    return;
+  }
+  
+  struct otc_publisher_callbacks publisher_callbacks = {0};
+  publisher_callbacks.user_data = session_data;
+  publisher_callbacks.on_stream_created = on_publisher_stream_created;
+  publisher_callbacks.on_render_frame = on_publisher_render_frame;
+  publisher_callbacks.on_stream_destroyed = on_publisher_stream_destroyed;
+  publisher_callbacks.on_error = on_publisher_error;
+  
+  session_data->publisher = otc_publisher_new("opentok-macos-sdk-samples",
+                                              NULL, /* Use WebRTC's video capturer. */
+                                              &publisher_callbacks);
+  
+  if (session_data->publisher == NULL) {
+    NSLog(@"Could not create OpenTok publisher successfully");
+    otc_session_delete(session_data->session);
+    return;
+  }
+}
+
+- (void)dealloc {
+  [self unsubscribe];
+  for (int i = 0; i < [_streamObjectList count]; i++)
+    otc_stream_delete([_streamObjectList pointerAtIndex:i]);
+  if (session_data->subscriber != NULL) {
+    otc_subscriber_delete(session_data->subscriber);
+    session_data->subscriber = NULL;
+  }
+  
+  [self unpublish];
+  if (session_data->publisher != NULL) {
+    otc_publisher_delete(session_data->publisher);
+    session_data->publisher = NULL;
+  }
+  
+  [self disconnect];
+  if (session_data->session != NULL) {
+    otc_session_delete(session_data->session);
+    session_data->session = NULL;
+  }
+    
+  free(session_data);
+  otc_destroy();
+}
+
+- (SessionData*)getSessionData {
+  return session_data;
+}
+
+- (void)connect {
+  if (session_data->session != NULL) {
+    otc_session_connect(session_data->session, TOKEN);
+  }
+}
+
+- (void)disconnect {
+  [self unpublish];
+  
+  if (session_data->session != NULL) {
+    otc_session_disconnect(session_data->session);
+  }
+}
+
+- (void)publish {
+  if ((session_data->session != NULL) && (session_data->publisher != NULL)) {
+      otc_session_publish(session_data->session, session_data->publisher);
+      
+      enable_tranformers(session_data->publisher);
```

Comment:
Just wondering to know if it's the same to apply transformers after or before publishing?

Reply:
at the moment it is not possible to apply the transformers before publishing